app/src/main/java/com/markkolenbrander/capstonenewsapp/adapters/ArticleAdapter.kt

```diff
@@ -0,0 +1,42 @@
+package com.markkolenbrander.capstonenewsapp.adapters
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.markkolenbrander.capstonenewsapp.models.Article
+import com.markkolenbrander.capstonenewsapp.views.ArticleView
+import com.markkolenbrander.capstonenewsapp.views.ArticleViewHolder
+
+class ArticleAdapter(
+    private val articlesList: ArrayList<Article?>,
+    private val onArticleTapped: (Article) -> Unit,
+) : RecyclerView.Adapter<ArticleViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ArticleViewHolder {
+        val articleView = ArticleView(parent.context)
+        articleView.layoutParams = ViewGroup.LayoutParams(
+            ViewGroup.LayoutParams.MATCH_PARENT,
+            ViewGroup.LayoutParams.WRAP_CONTENT,
+        )
+        return ArticleViewHolder(articleView)
+    }
+
+    override fun onBindViewHolder(holder: ArticleViewHolder, position: Int) {
+        holder.itemView.setOnClickListener {
+            articlesList[position]?.let { it1 -> onArticleTapped(it1) }
+        }
```

Comment:
This looks good to me! Nice work wiring up the click listener!

Only suggestion I'd make is provide a more verbose name rather than `it1`:

```suggestion
        holder.itemView.setOnClickListener {
            articlesList[position]?.let { article -> onArticleTapped(article) }
        }
```